File: tessoku/A05 - Three Cards/A05.ts
const mainA05 = (input: string): void => {
    const [N, K]: number[] = input.trim().split(' ').map(Number);
    let count: number = 0;
    for (let i: number = 1; i <= N; i++) {
        for (let j: number = 1; j <= N; j++) {
            let z: number = K - i - j;
            if (z >= 1 && z <= N) {
                count += 1;
            }
        }
    }

    console.log(count);
}

mainA05(require('fs').readFileSync('/dev/stdin', 'utf8'));

File: tessoku/A05 - Three Cards/A05.cs
using System;

class Program
{
   static void Main()
   {
       string[] input = Console.ReadLine().Split(' ');
       int N = int.Parse(input[0]);
       int K = int.Parse(input[1]);

       int count = 0;
       for (int i = 1; i <= N; i++)
       {
           for (int j = 1; j <= N; j++)
           {
               int z = K - i - j;
               if (z >= 1 && z <= N)
               {
                   count++;
               }
           }
       }

       Console.WriteLine(count);
   }
}